### DISCUSSION TITLE
Question: How to create another window in a desktop app?

### DISCUSSION THREAD

**Discussion by omnilogix** [microsoft/WindowsAppSDK#701] [Q&A] [answered]
When trying to create a second window, the following error is thrown:
_WinRT originate error - 0x80004005 : 'WinUI: Error creating second Desktop Window on the current process. No more than one Desktop Window is allowed per process.'._

This error contradicts the documentation here: https://docs.microsoft.com/en-us/windows/winui/api/microsoft.ui.xaml.window?view=winui-3.0

> In a Desktop app you can create each Window explicitly (more than one Window per thread is supported).

```
Window window = new Window();
window.Content = new TextBlock() { Text = "Hello" };
window.Activate();

```

> To create a new Window on a new thread in a Desktop app, create the thread first and then use the DispatcherQueueSynchronizationContext to queue work to another thread.
```
var thread = new Thread(() =>
{
    Window window = new Window();
    window.Content = new TextBlock() { Text = "Hello" };
    window.Activate();
    window.Closed += (_, __) => w.DispatcherQueue.InvokeShutdown();

    var syncContext = new DispatcherQueueSyncronizationContext();
    SynchronizationContext.SetSynchronizationContext(syncContext);

    this.ProcessEvents(); // Message pump
});

thread.SetApartmentState(ApartmentState.STA);
thread.Start();
```
Above code throws this error: 
_An exception of type 'System.Runtime.InteropServices.COMException' occurred in WinRT.Runtime.dll but was not handled in user code
The application called an interface that was marshalled for a different thread. (0x8001010E (RPC_E_WRONG_THREAD))_

Both of these examples throw exceptions.
**So, how do we create one or more additional windows in a WinUI 3.0 desktop app?**

**Comment by andrewleader** [comment, marked as answer]
Hi @omnilogix, WinUI 3.0 doesn't support multiple windows in the GA releases **yet**, but that's on their roadmap.

If you use the Preview versions of Project Reunion (we'll have another Preview release coming soon adding back preview features), you can start trying out multiple window support. You can find more info here: https://github.com/microsoft/microsoft-ui-xaml/issues/4709

**Reply by omnilogix** [reply]
@andrewleader - Thanks!   Any advice on whether to work with the preview releases, or adopt the GA's as the are made available? Or, maybe both?

**Reply by andrewleader** [reply]
Depending on your requirements, both 😜

If you need to ship your production app now using WinUI 3, you should use the GA releases. We do not promise support for the preview releases.

If you're not shipping till end of this year and are just trying to get a head start, you probably should just use **only the Preview releases**. Each GA release will remove the preview features that aren't shipping that release, so while you're still dependent on preview features (like multiple windows), I'd suggest skipping the GA's completely to avoid APIs that you were using disappearing, until the GA has multiple window support.

Or you can have one branch of your app on production GA releases, and one using the preview releases if you need to ship now while also playing with multiple window support 😊
